interop_task/interop_task.md

```diff
@@ -0,0 +1,123 @@
+| Proposal ID | TBC |
+|-------------|--------|
+| Name |  |
+| Date of Creation | 16 January 2019 |
+| Target | SYCL 1.2.1 extension |
+| Current Status | _Work in progress_ |
+| Reply-to | Victor Lomüller <victor@codeplay.com> |
+| Original author | Victor Lomüller <victor@codeplay.com>, Gordon Brown <gordon@codeplay.com>, Peter Zuzek <peter@codeplay.com> |
+| Contributors | Victor Lomüller <victor@codeplay.com>, Gordon Brown <gordon@codeplay.com>, Peter Zuzek <peter@codeplay.com> |
+
+# interop_task: Improving SYCL-OpenCL Interoperability
+
+## Motivation
+
+OpenCL applications often build around a set of of fixed function operations which take OpenCL buffers in order to run a hard-coded OpenCL kernels. However, because SYCL does not allow a user to access cl_mem object out of an cl::sycl::accessor, it difficult/impossible to integrate such library into a SYCL application, as the only current way to do this is to create all OpenCL buffers up-front, which is not always possible.
+
+This proposal introduces a way for a user to retrieve the OpenCL buffer associate with a SYCL buffer and enqueue a host task that can execute an arbitrary portion of host code within the SYCL runtime, therefore taking advantage of SYCL dependency analysis and scheduling.
+
+## Enqueuing host tasks on SYCL queues
+
+We introduce a new type of handler, the **codeplay::handler**, which includes a new
+**interop\_task** method that executes a task on the host.
+By submitting this command group to the SYCL device queue, we guarantee it is
+executed in-order w.r.t the other command groups on the same queue.
+Simultaneously, we guarantee that this operation is performed
+asynchronously w.r.t to the user-thread (therefore, enabling the user
+thread to continue submitting command groups).
+Other command groups enqueued in the same or different queues
+can be executed following the sequential consistency by guaranteeing the
+satisfaction of the requisites of this command group.
```

Comment:
I agree here that it should be the user's responsibility to avoid a race condition with the host or other interop operations, but there's shouldn't be any race conditions between an interop_task and other command groups because the data dependencies have to go through the SYCL runtime.